task.py

```diff
@@ -0,0 +1,114 @@
+"""Data format for Task."""
+
+
+class State(object):
+  UNUSED_STATUS = 0
+  PENDING = 1
+  FINISHED = 2
+  FAILED = 3
+  ABANDONED = 4
+
+
+class Task(object):
+  """Class for Task Object."""
+
+  def __init__(self, data=None):
+    """Initializes a task.
+
+    Args:
+      data: dictionary with all the task data (optional).
+    """
+    if data is None:
+      self._task = {}
+    else:
+      self._task = data
+
+  @property
+  def task_name(self):
+    return self._task['task_name']
+
+  @task_name.setter
+  def task_name(self, value):
+    """Sets name as string value."""
+    if isinstance(value, basestring):
+      self._task['task_name'] = value
+    else:
+      raise ValueError('Invalid input for task name: not a string')
+
+  @property
+  def dependent_on(self):
+    return self._task['dependent_on']
+
+  @dependent_on.setter
+  def dependent_on(self, value):
```

Comment:
perhaps better to have a add_depend_on() that takes a single object instead of an implicit list.